Meadowlark-js/packages/meadowlark-core/src/handler/NaturalKeyExtractor.ts

```diff
@@ -40,109 +41,114 @@ export type NaturalKeyWithSecurity = {
 /**
  * Takes a non-reference property representing a portion of the identity of a MetaEd entity,
  * an API JSON body matching that entity, and a path to the location of the property value
- * in the JSON body, and returns that portion of the natural key extracted from the JSON body.
+ * in the JSON body, and returns that portion of the document identity extracted from the JSON body.
  *
- * bodyPath is a path in the JSON body as a string array with one path segment per array element.
+ * documentPath is a path in the JSON body as a string array with one path segment per array element.
  */
-function singleNaturalKeyFrom(property: EntityProperty, body: object, bodyPath: string[]): NaturalKeyWithSecurity {
+function singleIdentityFrom(property: EntityProperty, body: object, documentPath: string[]): DocumentIdentityWithSecurity {
   const { apiMapping } = property.data.meadowlark as EntityPropertyMeadowlarkData;
-  const bodyPathAsString: string = [...bodyPath, apiMapping.fullName].join('.');
-  let naturalKeyValue: string | undefined = R.path([...bodyPath, apiMapping.fullName], body);
-  if (naturalKeyValue == null) naturalKeyValue = ''; // TODO: fatal error?
+  const documentPathAsString: string = [...documentPath, apiMapping.fullName].join('.');
+  const elementValue: string | undefined = R.path([...documentPath, apiMapping.fullName], body);
+
+  invariant(
+    elementValue != null,
+    `Identity element value for ${property.metaEdName} not found in ${JSON.stringify(body)} at ${documentPathAsString}`,
+  );
+
   return {
-    naturalKey: `${bodyPathAsString}=${naturalKeyValue}`,
-    studentId: apiMapping.fullName === 'studentUniqueId' ? naturalKeyValue : null,
```

Comment:
Similar changes here in identity (was natural key) extraction to above in reference (was foreign key) extraction.